src/scriptdev2/scripts/kalimdor/desolace.cpp

```diff
@@ -678,6 +678,74 @@ bool QuestAccept_npc_rigger_gizelton(Player* pPlayer, Creature* pCreature, const
     return true;
 }
 
+enum
+{
+    SPELL_CURSE_OF_MAGRAMI = 18159,
+    SPELL_GHOST_SPAWN_IN   = 17321,
+    SPELL_BLUE_AURA        = 17327,
+    SPELL_GREEN_AURA       = 18951,
+
+    FACTION_HOSTILE        = 16,
+};
+
+struct npc_magrami_spectre : public ScriptedAI
+{
+    npc_magrami_spectre(Creature* pCreature) : ScriptedAI(pCreature) { Reset(); }
+
+    uint32 m_uiCurseTimer;
+
+    void Reset() override
+    {
+        m_uiCurseTimer = 4000;
+    }
+
+    void JustAIInit() override
```

Comment:
Yes it is, but DoCastSpellIfCan contains redundant checks mostly done in Spell::CheckCast, hence if its not a combat spell, but a script spell, there is really no point in using it, ever since SD2 was merged with core.